home-work3/utils.js

```diff
@@ -0,0 +1,113 @@
+const csv = require('csv-parser');
+const fs = require('fs');
+const path = require('path');
+const jsonexport = require('jsonexport');
+
+const filePath = path.join('.', 'events-data.csv');
+
+const getEvents = async () => {
+    return new Promise((resolve, reject) => {
+        const results = [];
+        fs.createReadStream(filePath)
+            .on('error', (err) => {
+                reject(err)
+            })
+            .pipe(csv())
+            .on('data', (data) => results.push(data))
+            .on('end', () => {
+                resolve(results);
+            });
+    })
+};
+
+const getEventsByLocation = async (location) => {
+    return new Promise(((resolve, reject) => {
+        const data = [];
+        fs.createReadStream(filePath)
+            .on('error', (err) => {
+                reject(err)
+            })
+
+            .pipe(csv())
+            .on('data', (row) => {
+                if (location.toLowerCase() === row.Location.toLowerCase()) {
+                    data.push(row);
+                }
+            })
+
+            .on('end', () => {
+                resolve(data)
+            })
+    }))
+};
+
+const getEventById = async (id) => {
+    return new Promise(((resolve, reject) => {
+        let data;
+        fs.createReadStream(filePath)
+            .on('error', (err) => {
+                reject(err)
+            })
+
+            .pipe(csv())
+            .on('data', (row) => {
+                if (id.toString() === row.Id.toString()) {
+                    data = row;
+                    fs.createReadStream(filePath).close();
```

Comment:
With such implementation, we are creating a new read stream and closing it.

If we want to close stream initialized at line 47 we could write `this.close`. this should be set to the current stream. Or we could store a stream into some variable and call on it `.close`